coremltools/converters/nnssa/coreml/graph_pass/op_removals.py

```diff
@@ -70,23 +70,28 @@ def _remove_internal_identity_nodes(nnssa):
     delete_count = 0
     for fn_key in list(nnssa.functions.keys()):
         f = nnssa.functions[fn_key]
-        keys = list(f.graph.keys())
-        for k in keys:
-            if k not in f.graph:
+        for name in list(f.graph.keys()):
+            if name not in f.graph:
                 continue
-            node = f.graph[k]
-            if len(node.inputs) != 1 or len(node.outputs) != 1:
+            node = f.graph[name]
+
+            # Check if the node is in graph outputs
+            if len(node.inputs) != 1:
+                continue
+            if len(node.outputs) == 0 and len(node.control_outputs) == 0:
```

Comment:
In tf.2x, somewhat there are identity nodes which has empty outputs but non empty control_outputs.
So we should consider the identity as the output node only if it has both empty outputs and control_outputs.

Reply:
Is that easy to reproduce this case with a toy example?

Reply:
control dependency's input is a placeholder,
which only have control_outputs but not output haha.
I'll try to create a toy model for this haha.

Reply:
done haha